opentracing-api-extensions-tracer/src/main/java/io/opentracing/contrib/api/tracer/APIExtensionsSpan.java

```diff
@@ -0,0 +1,282 @@
+    }
+
+    @Override
+    public Span setBaggageItem(String name, String value) {
+        wrappedSpan.setBaggageItem(name, value);
+        if (!observers.isEmpty()) {
+            for (SpanObserver observer : observers) {
+                observer.onSetBaggageItem(this, name, value);
+            }
+        }
+        return this;
+    }
+
+    @Override
+    public Span log(Map<String, ?> fields) {
+        wrappedSpan.log(fields);
+        return handleLog(TimeUnit.MILLISECONDS.toMicros(System.currentTimeMillis()), fields);
+    }
+
+    @Override
+    public Span log(long timestampMicroseconds, Map<String, ?> fields) {
+        wrappedSpan.log(timestampMicroseconds, fields);
+        return handleLog(timestampMicroseconds, fields);
+    }
+
+    private Span handleLog(long timestampMicroseconds, Map<String, ?> fields) {
+        if (!observers.isEmpty()) {
+            for (SpanObserver observer : observers) {
+                observer.onLog(this, timestampMicroseconds, fields);
+            }
+        }
+        return this;
+    }
+
+    @Override
+    public Span log(String event) {
+        wrappedSpan.log(event);
+        return handleLog(TimeUnit.MILLISECONDS.toMicros(System.currentTimeMillis()), event);
+    }
+
+    @Override
+    public Span log(long timestampMicroseconds, String event) {
+        wrappedSpan.log(timestampMicroseconds, event);
+        return handleLog(timestampMicroseconds, event);
+    }
+
+    private Span handleLog(long timestampMicroseconds, String event) {
+        if (!observers.isEmpty()) {
+            for (SpanObserver observer : observers) {
+                observer.onLog(this, timestampMicroseconds, event);
+            }
+        }
+        return this;
+    }
+
+    @Override
+    public Span setTag(String key, String value) {
+        wrappedSpan.setTag(key, value);
+        return handleSetTag(key, value);
+    }
+    
+    @Override
+    public Span setTag(String key, boolean value) {
+        wrappedSpan.setTag(key, value);
+        return handleSetTag(key, value);
+    }
+
+    @Override
+    public Span setTag(String key, Number value) {
+        wrappedSpan.setTag(key, value);
+        return handleSetTag(key, value);
+    }
+
+    private Span handleSetTag(String key, Object value) {
+        tags.put(key, value);
+        if (!observers.isEmpty()) {
+            for (SpanObserver observer : observers) {
+                observer.onSetTag(this, key, value);
+            }
+        }
+        return this;
+    }
+
+    @Override
+    public Map<String,Object> getTags() {
+        return Collections.unmodifiableMap(tags);
+    }
+
+    @Override
+    public String getStringTag(String key) {
+        Object value = tags.get(key);
+        if (value instanceof String) {
+            return (String) value;
+        }
+        return null;
+    }
+
+    @Override
+    public Number getNumberTag(String key) {
+        Object value = tags.get(key);
+        if (value instanceof Number) {
```

Comment:
If I try to get a "int" from a string value, I would expect an exception, not null (at least, that's what happen in other APIs out there). If returning `null` is indeed what you think your callers would expect (or want), then please add a javadoc about it (sorry if it's added somewhere already)